pulsar-transformations/src/test/java/com/datastax/oss/pulsar/functions/transforms/ComputeStepTest.java

```diff
@@ -462,6 +497,181 @@ public static Object[] destinationTopicProvider() {
+  public static Object[][] primitiveTypeComputeProvider() {
+    return new Object[][] {
+      {
+        "oldValue",
+        Schema.STRING,
+        SchemaType.STRING,
+        "'newValue'",
+        ComputeFieldType.STRING,
+        "newValue",
+        Schema.STRING
+      },
+      {
+        "oldValue",
+        Schema.STRING,
+        SchemaType.STRING,
+        "fn:concat(value, '!')",
+        ComputeFieldType.STRING,
+        "oldValue!",
+        Schema.STRING
+      },
+      {
+        "1.3", Schema.STRING, SchemaType.STRING, "2.6", ComputeFieldType.DOUBLE, 2.6D, Schema.DOUBLE
+      },
+      {"3", Schema.INT32, SchemaType.INT32, "4", ComputeFieldType.INT32, 4, Schema.INT32},
+      {"3", Schema.INT32, SchemaType.INT32, "value + 2", ComputeFieldType.INT32, 5, Schema.INT32},
+      {
+        "3",
+        Schema.INT32,
+        SchemaType.INT32,
+        "'newValue'",
+        ComputeFieldType.STRING,
+        "newValue",
+        Schema.STRING
+      },
+      {"3", Schema.INT64, SchemaType.INT64, "4", ComputeFieldType.INT64, 4L, Schema.INT64},
+      {"3", Schema.INT64, SchemaType.INT64, "value + 2", ComputeFieldType.INT64, 5L, Schema.INT64},
+      {
+        "3",
+        Schema.INT64,
+        SchemaType.INT64,
+        "'newValue'",
+        ComputeFieldType.STRING,
+        "newValue",
+        Schema.STRING
+      },
+      {"3.2", Schema.FLOAT, SchemaType.FLOAT, "3.3", ComputeFieldType.FLOAT, 3.3F, Schema.FLOAT},
+      {
+        "3.2",
+        Schema.FLOAT,
+        SchemaType.FLOAT,
+        "value + 1",
+        ComputeFieldType.FLOAT,
+        4.2F,
+        Schema.FLOAT
+      },
+      {
+        "3.2",
+        Schema.FLOAT,
+        SchemaType.FLOAT,
+        "'newValue'",
+        ComputeFieldType.STRING,
+        "newValue",
+        Schema.STRING
+      },
+      {
+        "3.2", Schema.DOUBLE, SchemaType.DOUBLE, "3.3", ComputeFieldType.DOUBLE, 3.3D, Schema.DOUBLE
+      },
+      {
+        "3.2",
+        Schema.DOUBLE,
+        SchemaType.DOUBLE,
+        "value + 1",
+        ComputeFieldType.DOUBLE,
+        4.2D,
+        Schema.DOUBLE
+      },
+      {
+        "3.2",
+        Schema.DOUBLE,
+        SchemaType.DOUBLE,
+        "'newValue'",
+        ComputeFieldType.STRING,
+        "newValue",
+        Schema.STRING
+      },
+      {
+        "false",
+        Schema.BOOL,
+        SchemaType.BOOLEAN,
+        "true",
+        ComputeFieldType.BOOLEAN,
+        true,
+        Schema.BOOL
+      },
+      {
+        "false",
+        Schema.BOOL,
+        SchemaType.BOOLEAN,
+        "value || true",
+        ComputeFieldType.BOOLEAN,
+        true,
+        Schema.BOOL
+      },
+      {"true", Schema.BOOL, SchemaType.BOOLEAN, "1", ComputeFieldType.DOUBLE, 1D, Schema.DOUBLE},
+      {
+        "2007-01-02",
+        Schema.DATE,
+        SchemaType.DATE,
+        "'2008-02-07'",
+        ComputeFieldType.DATE,
+        LocalDate.parse("2008-02-07"),
+        Schema.DATE
```

Comment:
This schema and value are not compatible.
Schema should be `Schema.LOCAL_DATE` or value should be a `java.util.Date`

Reply:
let me resolve the discrepancies in Pulsar API - they directly map to the java types and it makes sense for them to match. I'll also check the exhaustive list of schemas and try to add them (like byes) if they don't blow up the scope of this patch...

Reply:
Also `j.u.Date` is deprecated so implementation wise we're using the `LocalDate` - does the Java type returned by the API has to strictly map to the Pulsar Schema API? 

I'm more more thinking now that the Compute field types should actually be our own (not 1:1 mapping with Pulsar API) so we have more control over our implementation and can always document the behavior for users...

Reply:
> does the Java type returned by the API has to strictly map to the Pulsar Schema API

Yes, Pulsar Schema.DATE can only encode/decode j.u.Date.
I guess we could decide to not support older Java types in `compute`. And only have them in `cast`.
My concern is that from a user perspective there will be a doubt as to whether DATE maps to Date or LocalDate.

Reply:
Makes sense. I'll stick to `java.time.*`.

Reply:
Previously, we chose `j.t.OffsetDateTime` over `j.t.LocalDateTime` as a better representation of `RFC3339` dates (e.g. it doesn't parse if there is `Z ` in the end). The pulsar schema API thought doesn't have a Schema.OffsetDateTime though. I wonder if in this case it is Ok to return `j.t.OffsetDateTime` or just switch to `j.t.LocalDateTime`

Reply:
So I kept `OffsetDateTime` to capture input, and switched output to `Instant` - it seems the only schema time available in the public API today that can represent full RFC3339 date/time.